### PR TITLE
Basic Keychain storage

### PR DESCRIPTION
**Pull request by maksymmalyhin**
- read-write-remove an object
- tests

### CODE REVIEW

**Comment by paulb777** [comment]
Why does pod lib lint fail on the PR CI run, but not on the branch CI run?

File: FirebaseInstallations.podspec [hunk 1]
```diff
@@ -26,6 +26,7 @@ Pod::Spec.new do |s|
 
     base_dir = "FirebaseInstallations/Source/"
     s.source_files = base_dir + '**/*.[mh]'
```
Comment (paulb777) [inline FirebaseInstallations.podspec]: Make a directory like `Library` and delete the `exclude_files`

File: FirebaseInstallations/Source/FIRSecureStorage.h [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface FIRSecureStorage : NSObject
+
+- (instancetype)initWithService:(NSString *)service;
+
+- (FBLPromise<id<NSSecureCoding>> *)getObjectForKey:(NSString *)key
+                                        objectClass:(Class)objectClass
+                                        accessGroup:(nullable NSString *)accessGropup;
```
Comment (paulb777) [inline FirebaseInstallations/Source/FIRSecureStorage.h]: typo

File: FirebaseInstallations/Source/FIRSecureStorage.h [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+- (FBLPromise<NSNull *> *)setObject:(id<NSSecureCoding>)object
+                             forKey:(NSString *)key
+                        accessGroup:(nullable NSString *)accessGropup;
```
Comment (paulb777) [inline FirebaseInstallations/Source/FIRSecureStorage.h]: typo

File: FirebaseInstallations/Source/FIRSecureStorage.h [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+- (FBLPromise<NSNull *> *)removeObjectForKey:(NSString *)key
+                                 accessGroup:(nullable NSString *)accessGropup;
```
Comment (paulb777) [inline FirebaseInstallations/Source/FIRSecureStorage.h]: typo

File: FirebaseInstallations/Source/FIRSecureStorage.m [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+                          do:^id {
+                            NSDictionary *query = [self keychainQueryWithKey:key
+                                                                 accessGroup:accessGropup];
+                            NSError *error;
+                            NSData *encodedObject = [self getItemWithQuery:query error:&error];
+
+                            if (error) {
+                              return error;
+                            }
+
```
Comment (paulb777) [inline FirebaseInstallations/Source/FIRSecureStorage.m]: Too many blank lines?

**Comment by maksymmalyhin** [comment]
@paulb777 Those are different versions of code. The branch build was for merge `master` into `fis-master`. I am looking into the PR failure...

File: FirebaseInstallations/Source/Library/FIRSecureStorage.m [hunk 6]
```diff
@@ -0,0 +1,272 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import "FIRSecureStorage.h"
+#import <Security/Security.h>
+
+#if __has_include(<FBLPromises/FBLPromises.h>)
```
Comment (maksymmalyhin) [inline FirebaseInstallations/Source/Library/FIRSecureStorage.m]: I haven't found out why Firebase imports work as is but FBLPromises doesn't. I'll investigate it. Any suggestions are appreciated.

Comment (paulb777) [reply FirebaseInstallations/Source/Library/FIRSecureStorage.m]: Weird CocoaPods should always make the module syntax work.

Comment (maksymmalyhin) [reply FirebaseInstallations/Source/Library/FIRSecureStorage.m]: I suspect, there might be some details with https://github.com/google/promises/blob/master/PromisesObjC.podspec I'll make sure it's resolved before we publish the pod

File: FirebaseInstallations/Source/FIRSecureStorage.m (hunk 5 above)
Comment (maksymmalyhin) [reply FirebaseInstallations/Source/FIRSecureStorage.m]: I'll remove some in the next PRs if you don't mind

**Pull request merged**